### PR TITLE
expected_total_number_of_files_per_rse and rules_with_0_completion_vo…

### PR DESCRIPTION
**Pull request by FernandoGarzon**
…lume ported to SQLAlchemy

### CODE REVIEW

File: common/check_number_of_missing_files_per_rse
Comment (nsmith-) [inline common/check_number_of_missing_files_per_rse:46]: Probably only need one of these lines

File: common/check_expected_total_number_of_files_per_rse
Comment (nsmith-) [inline common/check_expected_total_number_of_files_per_rse:38]: This is counting the number of files that are part of an incomplete rule, grouped by the _RSE expression_ rather than by _RSE_. This seems useful but should be named appropriately

File: common/check_number_of_missing_files_per_rse
Comment (nsmith-) [inline common/check_number_of_missing_files_per_rse:49]: Is there any purpose to `status = CRITICAL` ? Looks like it is ignored. We would have to define thresholds and this doesn't appear to be a good place to do so.

File: common/check_report_used_space
Comment (nsmith-) [inline common/check_report_used_space:30]: We should record a gauge per usage source per RSE

File: common/check_rules_with_0_completion_volume
Comment (nsmith-) [inline common/check_rules_with_0_completion_volume:48]: It's not clear to me what the purpose of this probe is, given we have already a probe counting the stuck volume.

File: common/check_space_uniquely_used_by_rucio
Comment (nsmith-) [inline common/check_space_uniquely_used_by_rucio:58]: This is a CMS-specific probe, is there a better place to put it?

Comment (ericvaandering) [reply common/check_space_uniquely_used_by_rucio:58]: Yeah, I think we should make our own repository of probes. I can pick it up from this branch for the moment.

File: common/check_space_uniquely_used_by_rucio
Comment (ericvaandering) [inline common/check_space_uniquely_used_by_rucio:43]: There is an error here. You are closing the last ( too soon) should be after line 45.

File: common/check_space_uniquely_used_by_rucio
Comment (ericvaandering) [reply common/check_space_uniquely_used_by_rucio:58]: Please remove this from this commit and make a PR against CMSRucio/probes

**Comment by ericvaandering** [comment]
@FernandoGarzon can you address Nick's comments and mine? I'd like to get this merged in early in the new year. I've let this go too long. :-(

File: common/check_number_of_missing_files_per_rse
Comment (FernandoGarzon) [reply common/check_number_of_missing_files_per_rse:48]: What do you propose to do here? Should I move or remove this line? Where should it be moved, exactly?

File: common/check_number_of_missing_files_per_rse
Comment (FernandoGarzon) [reply common/check_number_of_missing_files_per_rse]: This is addressed.

File: common/check_expected_total_number_of_files_per_rse
Comment (FernandoGarzon) [reply common/check_expected_total_number_of_files_per_rse]: This is also addressed.

File: common/check_report_used_space
Comment (FernandoGarzon) [reply common/check_report_used_space:30]: I don't get what you mean by 'gauge per usage source per RSE'. How's that different from what there is already in this probe? 
https://github.com/rucio/rucio/blob/master/lib/rucio/core/rse.py#L680

File: common/check_rules_with_0_completion_volume
Comment (FernandoGarzon) [reply common/check_rules_with_0_completion_volume:48]: @ericvaandering should we delete this probe?

File: common/check_number_of_missing_files_per_rse
Comment (nsmith-) [reply common/check_number_of_missing_files_per_rse:48]: I think the line could be removed

File: common/check_report_used_space
Comment (nsmith-) [reply common/check_report_used_space:30]: There are some usages with `usage['source']` different from `rucio`, we should report those as well, e.g. `expired` and `obsolete`. I would make new stats for each.

File: common/check_space_uniquely_used_by_rucio
Comment (nsmith-) [reply common/check_space_uniquely_used_by_rucio]: Now that phedex is retired, this is essentially obsolete

Comment (FernandoGarzon) [reply common/check_space_uniquely_used_by_rucio]: This probe was deleted from PR.

File: common/check_number_of_missing_files_per_rse
Comment (FernandoGarzon) [reply common/check_number_of_missing_files_per_rse:48]: Line removed

File: common/check_report_used_space
Comment (FernandoGarzon) [reply common/check_report_used_space]: Stats were added.

File: common/check_rules_with_0_completion_volume
Comment (FernandoGarzon) [reply common/check_rules_with_0_completion_volume:48]: I'm taking out this file from this commit.

**Comment by ericvaandering** [comment]
@dchristidis can you also check this. These are probes CMS is using which we feel would be useful in general.

**Comment by dchristidis** [comment]
Thank you @FernandoGarzon. Please place the new probes into a `cms` directory (instead of `common`).

Also, the commit messages are misleading. Please rewrite your branch (one commit with ‘added new CMS probes’ or equivalent might be sufficient).

**Comment by FernandoGarzon** [comment]
@dchristidis Your suggestions were taken into account. All the files are on the cms/ folder and this PR consists only of one commit. The final number of files to merge is 3. 

Thank you.

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (dchristidis) [inline cms/check_expected_total_number_of_files_per_rse_expression:44]: I’m not a huge fan of not indenting this multi-line statement, but this is personal opinion.

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (dchristidis) [inline cms/check_expected_total_number_of_files_per_rse_expression:48]: I see two potential issues with this:
 1. RSE expressions can be complex, such as `(istape=True&Tier=1)\site=IN2P3-CC)`.
 2. Once a statistic is defined, its value will forever remain above `0`.

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (dchristidis) [inline cms/check_expected_total_number_of_files_per_rse_expression:50]: Please remove the space.

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (dchristidis) [inline cms/check_expected_total_number_of_files_per_rse_expression:53]: Please remove the empty last line.

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (dchristidis) [inline cms/check_expected_total_number_of_files_per_rse_expression:52]: If `status` was set to `CRITICAL`, the probe will still exit with `OK`.

File: cms/check_number_of_missing_files_per_rse
Comment (dchristidis) [inline cms/check_number_of_missing_files_per_rse:50]: Please add a space after the comma.

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (dchristidis) [inline cms/check_expected_total_number_of_files_per_rse_expression:34]: Best to be consistent with the use of single or double quotes.

**Review by dchristidis — COMMENTED** [review]
I’ve made some comments. Most of them are of an aesthetic nature; you may regard them as optional but recommended. If you decide to make changes, please amend this commit instead of adding more.

Note that I’ve made my comments at the first occurrence of an issue, but most of them apply to all files and/or multiple times in the same file.

My review won’t block merging this PR, but someone else from CMS must approve it.

**Comment by FernandoGarzon** [comment]
@dchristidis most of your comments were taken into account. I believe these are good enough to be merged.

**Comment by dchristidis** [comment]
My apologies, I think one of my comments was miscommunicated: I suggested removing the empty newline from one of the files, not all newline characters at the end of the last non-empty line.

Anyway, I am still waiting for an explicit approval. @ericvaandering? @nsmith-?

File: cms/check_expected_total_number_of_files_per_rse_expression
Comment (nsmith-) [reply cms/check_expected_total_number_of_files_per_rse_expression:50]: I'm also a bit worried about the multiplicity of such expressions. I think in our case, since we immediately translate the statsd gauges to prometheus metrics we avoid the permanent >0 issue by letting those go null. So I think we can leave it for now.

**Review by nsmith- — APPROVED** [review]
Looks good.

**Comment by dchristidis** [comment]
Thanks!

**Pull request merged**
